src/kernel/arch/x86/syscalls.zig

```diff
@@ -0,0 +1,248 @@
+///     IN handler: SyscallHandler - The handler to register the syscall with
+///
+/// Errors:
+///     SyscallError.InvalidSyscall - If the syscall is invalid (see isValidSyscall)
+///     SyscallError.SyscallExists - If the syscall has already been registered
+///
+pub fn registerSyscall(syscall: u8, handler: SyscallHandler) !void {
+    if (!isValidSyscall(syscall))
+        return SyscallError.InvalidSyscall;
+    if (handlers[syscall]) |_|
+        return SyscallError.SyscallExists;
+    handlers[syscall] = handler;
+}
+
+inline fn syscall0(syscall: u32) u32 {
+    return asm volatile (
+        \\int $0x80
+        : [ret] "={eax}" (-> u32)
+        : [syscall] "{eax}" (syscall),
+    );
+}
+
+inline fn syscall1(syscall: u32, arg: u32) u32 {
+    return asm volatile (
+        \\int $0x80
+        : [ret] "={eax}" (-> u32)
+        : [syscall] "{eax}" (syscall),
+          [arg1] "{ebx}" (arg),
+    );
+}
+
+inline fn syscall2(syscall: u32, arg1: u32, arg2: u32) u32 {
+    return asm volatile (
+        \\int $0x80
+        : [ret] "={eax}" (-> u32)
+        : [syscall] "{eax}" (syscall),
+          [arg1] "{ebx}" (arg1),
+          [arg2] "{ecx}" (arg2),
+    );
+}
+
+inline fn syscall3(syscall: u32, arg1: u32, arg2: u32, arg3: u32) u32 {
+    return asm volatile (
+        \\int $0x80
+        : [ret] "={eax}" (-> u32)
+        : [syscall] "{eax}" (syscall),
+          [arg1] "{ebx}" (arg1),
+          [arg2] "{ecx}" (arg2),
+          [arg3] "{edx}" (arg3),
+    );
+}
+
+inline fn syscall4(syscall: u32, arg1: u32, arg2: u32, arg3: u32, arg4: u32) u32 {
+    return asm volatile (
+        \\int $0x80
+        : [ret] "={eax}" (-> u32)
+        : [syscall] "{eax}" (syscall),
+          [arg1] "{ebx}" (arg1),
+          [arg2] "{ecx}" (arg2),
+          [arg3] "{edx}" (arg3),
+          [arg4] "{esi}" (arg4),
+    );
+}
+
+inline fn syscall5(syscall: u32, arg1: u32, arg2: u32, arg3: u32, arg4: u32, arg5: u32) u32 {
+    return asm volatile (
+        \\int $0x80
+        : [ret] "={eax}" (-> u32)
+        : [syscall] "{eax}" (syscall),
+          [arg1] "{ebx}" (arg1),
+          [arg2] "{ecx}" (arg2),
+          [arg3] "{edx}" (arg3),
+          [arg4] "{esi}" (arg4),
+          [arg5] "{edi}" (arg5),
+    );
+}
+
+///
+/// Gets the syscall argumentaccording to the given number. 0 => ebx, 1 => ecx, 2 => edx, 3 => esi and 4 => edi
+///
+/// Arguments:
+///     IN ctx: *arch.InterruptContext - The interrupt context from which to get the argument
+///     IN num: comptime u32 - The arg number to get. Between 0 and 4.
+///
+pub inline fn syscallArg(ctx: *arch.InterruptContext, comptime num: u32) u32 {
+    return switch (num) {
+        0 => ctx.ebx,
+        1 => ctx.ecx,
+        2 => ctx.edx,
+        3 => ctx.esi,
+        4 => ctx.edi,
+        else => @compileError("Arg number must be between 0 and 4")
+    };
+}
+
+pub inline fn syscallArg2(ctx: *arch.InterruptContext) u32 {
```

Comment:
Where syscallArg1? Also as you have syscallArg where you can index all the argument, do you need all these functions?

Reply:
Ah thanks for spotting this. I wrote `syscallArgX` before writing the general one and forgot to remove the old ones.